module/VuFind/src/VuFind/Controller/AlmaController.php

```diff
@@ -0,0 +1,342 @@
+            default:
+            	$accessPermission = 'access.alma.webhook.challenge';
+            	try {
+					$this->checkPermission($accessPermission);
+				} catch (\VuFind\Exception\Forbidden $ex) {
+					return $this->createJsonResponse('Access to Alma Webhook challenge forbidden. Set permission \''.$accessPermission.'\' in \'permissions.ini\'.', 403);
+				}
+                return $this->webhookChallenge();
+                break;
+        }  
+    }
+    
+    /**
+     * Webhook actions related to a newly created, updated or deleted user in Alma.
+     * 
+     * @param mixed $requestBodyJson		A JSON string decode with json_decode()
+     * @return NULL|\Zend\Http\Response
+     */
+    protected function webhookUser($requestBodyJson) {
+    	
+    	// Initialize user variable that should hold the user table row
+        $user = null;
+        
+        // Initialize response variable
+        $jsonResponse = null;
+        
+        // Get method from webhook (e. g. "create" for "new user")
+        $method = (isset($requestBodyJson->webhook_user->method)) ? $requestBodyJson->webhook_user->method : null;
+        
+        // Get primary ID
+        $primaryId = (isset($requestBodyJson->webhook_user->user->primary_id)) ? $requestBodyJson->webhook_user->user->primary_id : null;
+        
+        if ($method == 'CREATE' || $method == 'UPDATE') {
+            // Get username (could e. g. be the barcode)
+            $username = null;
+            $userIdentifiers = (isset($requestBodyJson->webhook_user->user->user_identifier)) ? $requestBodyJson->webhook_user->user->user_identifier : null;
+            $idTypeConfig = ($this->configAlma->NewUser->idType != null && isset($this->configAlma->NewUser->idType)) ? $this->configAlma->NewUser->idType : null;
+            foreach ($userIdentifiers as $userIdentifier) {
+                $idTypeHook = (isset($userIdentifier->id_type->value)) ? $userIdentifier->id_type->value : null;
+                if ($idTypeHook != null && $idTypeHook == $idTypeConfig && $username == null) {
+                    $username = (isset($userIdentifier->value)) ? $userIdentifier->value : null;
+                }
+            }
+            
+            // Use primary ID as username as a fallback if no other username ID is available
+            $username = ($username == null) ? $primaryId : $username;
+            
+            // Get user details from Alma Webhook message
+            $firstname = (isset($requestBodyJson->webhook_user->user->first_name)) ? $requestBodyJson->webhook_user->user->first_name : null;
+            $lastname = (isset($requestBodyJson->webhook_user->user->last_name)) ? $requestBodyJson->webhook_user->user->last_name : null;
+            
+            $allEmails = (isset($requestBodyJson->webhook_user->user->contact_info->email)) ? $requestBodyJson->webhook_user->user->contact_info->email : null;
+            $email = null;
+            foreach ($allEmails as $currentEmail) {
```

Comment:
There are some formatting issues here, many of which can probably be addressed automatically by php-cs-fixer. Let me know if you need any help/guidance on getting them sorted out so that Travis builds stop failing.